OpenRA.Mods.Common/Traits/Production.cs

```diff
@@ -23,14 +23,17 @@ public class ProductionInfo : PausableConditionalTraitInfo
 		[Desc("e.g. Infantry, Vehicles, Aircraft, Buildings")]
 		public readonly string[] Produces = Array.Empty<string>();
 
+		[Desc("When owner is changed, should the Faction be updated to the new owner's faction?")]
```

Comment:
I think this should explain what "faction" really means. Afaik it changes so that the produced units are of your own faction, meaning that if you capture an enemy war factory you'd no longer be able to build an MCV of their faction?

Reply:
But in base openra the bug is just about the UI no? It's something that should be fixed in TS as well

Reply:
Again it depends on the mod. For ORA default mods (TS too) I assume the behavior needs to stay the same (i.e. `Faction` does not change after owner change), hence the switch defaults to false.

Also it's not just UI thing, there are other places, where `Production.Faction` is used, such as `PlaceBuildingOrderGenerator` (in `VariantWrapper`) or `*Production` classes.

I already had a UI-only solution for this bug (#20705), but that was rejected.

Reply:
@PunkPun if you capture a Nod war factory in TS you will want to be able to build Nod units.

Reply:
Hence, this is (`Production.UpdateFactionOnOwnerChange`) entirely optional and so is the `Sticky` flag on `ProductionQueue`. I can rename `Sticky` flag, but that would require update rule.

Reply:
Actually I hope this bug can be fixed in this PR, for #21162

>This issue (Production UI upgrade) has some very nasty side effect on fixing.

>You need a test on owner change before/after disable or enable. There can be some bugs that owner changed facility can still produce on old owner's production UI.

It is not only the problem of faction change, it is the problem of when you losing control of production actor it is still on your production UI and you can produce unit on that actor.